### PR TITLE
Separate out react components

### PR DESCRIPTION
**Pull request by iHiD**
This separates out React components from (Rails) view components.

I've done this because their bases have diverged. The rails view components now need to be called with `render ViewComponents::...` in order for the context to be passed in. The React Components can stay as more "dumb" components as they do not require the extra functionality of the view context.

I think I've got it all right, but a sanity check eyeballing would be appreciated.

---

This also adds the "Join" button at the top-right of this, which is the thing that required the view context (to set the csrf token correctly, and lead me to shave this yak)

Screenshot 2020-11-11 at 02 19 18

### CODE REVIEW

File: app/helpers/view_components/view_component.rb [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
+      :tag, :link_to, :button_to,
+      :graphical_icon, :icon, :track_icon, :exercise_icon,
+      to: :view_context
+
+    # This is called when you called `render SomeComponent.new(...)`
+    # It sets the view context, which can then be used for things
+    # like getting the current user, or authentication tokens
+    def render_in(context, *_args)
+      @view_context = context
+      to_s
```
Comment (iHiD) [inline app/helpers/view_components/view_component.rb:17]: This whole PR is basically to add these lines.

Comment (kntsoriano) [reply app/helpers/view_components/view_component.rb:17]: Didn't know this. Nice!

**Review by kntsoriano — APPROVED** [review]

**Review by ErikSchierboom — APPROVED** [review]
LGTM

File: app/helpers/view_components/view_component.rb (hunk 1 above)
Comment (iHiD) [reply app/helpers/view_components/view_component.rb:17]: Me neither. Reading the Rails source code++

**Pull request merged**